tuist/tuist
# Fix scheme archive action not respecting custom configurations

Resolves: https://github.com/tuist/tuist/issues/6621

### Short description 📝

- In the event a custom defined scheme's archive action is specified without a build action it was getting ignored
- The scheme generation relied on a build action's data to attempt to generate a default / fallback archive action in case an archive action was not explicitly specified
- This logic was mistakingly returning early in case a build action wasn't specified (as this code path should only be exercised in the event an archive action isn't explicitly being specified)
- This has now been corrected
- Tests and fixtures were updated accordingly


### How to test the changes locally 🧐

- Generate the fixture `ios_app_with_multi_configs`

```
swift build
swift run tuist generate --path fixtures/ios_app_with_multi_configs
```

- Verify the scheme `App-Beta-ArchiveOnly` has an archive action with the configuration `Beta`


### Contributor checklist ✅


### Reviewer checklist ✅
